### PR TITLE
Various corrections in the docs

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

I have gone through all compositions except Alibaba provider because I don't have creds there. I did a pass on OAM docs, too, using AWS VPC composition and it works as expected.

Note that I didn't change the CLI installation commands and provider installation with Crossplane Helm charts as these are not ready yet. 

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Manually.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/introduction/providers.md [hunk 1]
```diff
@@ -39,18 +39,17 @@ Here is the list of current providers:
...
 
 ```yaml
-apiVersion: packages.crossplane.io/v1alpha1
-kind: ClusterPackageInstall
+apiVersion: pkg.crossplane.io/v1alpha1
+kind: Provider
 metadata:
-  name: provider-gcp
-  namespace: crossplane-system
+  name: provider-aws
```
Comment (hasheddan) [inline docs/introduction/providers.md:50]: any particular reason for updating all of the examples to be AWS?

File: docs/introduction/managed-resources.md [hunk 2]
```diff
@@ -81,12 +81,12 @@ be provided when a new version of that CRD schema is released.
 
 In general, managed resources are high fidelity resources meaning they will provide
 parameters and behaviors that are provided by the external resource API. This applies
-to grouping of resources, too. For example, `RDSInstance` appears under `database`
-API group, so, its `APIVersion` and `Kind` look like the following:
+to grouping of resources, too. For example, `Queue` appears under `sqs` API group
+in AWS,so, its `APIVersion` and `Kind` look like the following:
```
Comment (hasheddan) [inline docs/introduction/managed-resources.md:85]: Is this change because RDS shouldn't really be in the `database` group? :sweat_smile:

**Review by hasheddan — APPROVED** [review]
Two minor comments, but LGTM!

File: docs/introduction/managed-resources.md (hunk 2 above)
Comment (muvaf) [reply docs/introduction/managed-resources.md:85]: Hahah yes, shhh 😄

File: docs/introduction/providers.md (hunk 1 above)
Comment (muvaf) [reply docs/introduction/providers.md:50]: One of them was GCP and other one Azure. I just wanted both to be the same, no particular concerns.

**Pull request merged**